Dynatrace/dynatrace-operator
# Propagate namespace label references as environment variables used for version detection

# Description
Namespace can be annotated to add additional mapping:
```
annotations:
  mapping.release.dynatrace.com/version: "metadata.annotations['my-version']"
  mapping.release.dynatrace.com/product: "metadata.labels['app.kubernetes.io/name']"
  mapping.release.dynatrace.com/stage: "metadata.namespace"
  mapping.release.dynatrace.com/build-version: "metadata.labels['release.dynatrace.com/stage']"
```
This will configure the mapping used to get the value of the corresponding environment variable via reference in the injected pod.
References are not validated by operator.

Mapping for the different environment variables:
- `DT_RELEASE_VERSION`
  - mapping: `mapping.release.dynatrace.com/version`
  - fallback to default when mapping is missing
- `DT_RELEASE_PRODUCT`
  - mapping: `mapping.release.dynatrace.com/product`
  - fallback to default when mapping is missing
- `DT_RELEASE_STAGE`
  - mapping: `mapping.release.dynatrace.com/stage`
  - no default, no environment variable is missing
- `DT_RELEASE_BUILD_VERSION`
  - mapping: `mapping.release.dynatrace.com/build-version`
  - no default, no environment variable is missing

## How can this be tested?
Apply dynakube CR with `feature.dynatrace.com/label-version-detection: "true"`
Annotate monitored apps namespace with mappings described above, for example:
```
apiVersion: v1                                                                     
kind: Namespace                                                                    
metadata:                                                                          
  annotations:                                                                     
    mapping.release.dynatrace.com/build-version: metadata.annotations['my-build-version']
    mapping.release.dynatrace.com/product: metadata.annotations['my-product']      
    mapping.release.dynatrace.com/stage: metadata.annotations['my-stage']        
    mapping.release.dynatrace.com/version: metadata.annotations['my-version']      
```
Create pod with annotations set:
```
apiVersion: v1                                                 
kind: Pod                                                      
metadata:                                                      
  annotations:                                                 
    data-ingest.dynatrace.com/injected: "true"                 
    dynakube.dynatrace.com/injected: "true"                    
    my-build-version: my custom build-version from deployment-pod-template 
    my-product: ppp
    my-stage: sss
    my-version: bbb
```
Check if pod's env's env variables have values taken from pod's annotations

## Checklist
